### PR TITLE
fix #92

### LINKED ISSUE #92: Emulate packets open.mp read offset
see `legacy_network_impl.cpp` 

https://github.com/openmultiplayer/open.mp/blob/8c246565edd6363ee55dfd01043f5c0c48f92f09/Server/Components/LegacyNetwork/legacy_network_impl.cpp#L917

```c++
bs.SetReadOffset(8); // Ignore packet ID
return handler->onReceivePacket(*player, type, bs);
```

but in `script.cc`, line 142, 155, 190, 203
https://github.com/katursis/Pawn.RakNet/blob/d2fef5af377ad090dd47908e13f170da33983119/src/script.cc#L140

```c++
bs->resetReadPointer(); // forget ignore packet ID ? --- bs->SetReadOffset(8); 
return handler->onReceivePacket(*player, packet_id, *bs);
```

`handler->onReceivePacket and handler->onReceive`

test it this way
1. connectNpc(foo), and assume that the id of this npc is 0
2. emulate an onFootSync packet and call the PR_EmulateIncomingPacket, params set playerid to 0.
3. get the npc position, if they are approach to where you just set.

### PR DESCRIPTION
**Pull request by dockfries**
let's go!

### CODE REVIEW

**Comment by katursis** [comment]
Thank you for your contribution.

1) I'm concerned about old commits being present. Let's fix it so this PR only contains your changes.
2) Have you personally tested that the emulation works as expected with your changes?
3) I think the proposed changes should not affect PR_EmulateIncomingRPC, because BitStream RPC doesn't have a packetid at the beginning.
4) The bs->resetReadPointer() call can be removed if bs->SetReadOffset(8) is added.

**Comment by dockfries** [comment]
1. I will reset the commit and recommit it.
2. I tested in my environment (samp-node) only packet, it works, and I believe that's the issue.
3. for rpc, i'm wrong, yes , no `SetReadOffset` is needed.
4. ok, removed `resetReadPointer()`

---
I couldn't find the source code about `onReceivePacket` (only `network.hpp`), and refer to `legacy_network_impl.cpp`
I just saw that open.mp calls `SetReadOffset(8)`.

File: src/script.cc [hunk 1]
```diff
@@ -137,8 +137,7 @@ cell Script::PR_EmulateIncomingPacket(BitStream *bs, int player_id) {
                 return true;
               }
 
-              bs->resetReadPointer();
-
```
Comment (katursis) [inline src/script.cc:141]: Line breaks were removed by mistake, let's put them back on lines 141, 154, 189, 202.

Comment (dockfries) [reply src/script.cc:141]: ok

**Pull request merged**